frontend/src/test/scala/bloop/tasks/TaskTest.scala

```diff
@@ -1,121 +1,127 @@
-      addThree.dependsOn(addOne)
-      addThree.dependsOn(addTwo)
+  @Test
+  def twoRootsOneDependent = {
+    val buf = Buffer.empty[Int]
+    def add(x: Int): Task[Unit] = new Task(_ => buf.synchronized { buf += x; () }, () => ())
 
-      addThree.run().map { _ =>
-        assert(buf == Buffer(1, 2, 3) || buf == Buffer(2, 1, 3))
-      }
-    }
+    val addOne = add(1)
+    val addTwo = add(2)
+    val addThree = add(3)
+    addThree.dependsOn(addOne)
+    addThree.dependsOn(addTwo)
 
-    "Un-necessary tasks are not run" - {
-      val buf = Buffer.empty[Int]
-      def add(x: Int): Task[Unit] = new Task(_ => buf += x, () => ())
+    val _ = run(addThree)
+    assertTrue(s"buf = $buf", buf == Buffer(1, 2, 3) || buf == Buffer(2, 1, 3))
+  }
 
-      val addOne = add(1)
-      val addTwo = add(2)
-      val addThree = add(3)
-      val addFour = add(4)
+  @Test
+  def unnecessaryTasksAreNotRun = {
+    val buf = Buffer.empty[Int]
+    def add(x: Int): Task[Unit] = new Task(_ => buf.synchronized { buf += x; () }, () => ())
 
-      addTwo.dependsOn(addOne)
-      addThree.dependsOn(addTwo)
-      addFour.dependsOn(addThree)
+    val addOne = add(1)
+    val addTwo = add(2)
+    val addThree = add(3)
+    val addFour = add(4)
 
-      addThree.run().map { case _ => assert(buf == Buffer(1, 2, 3)) }
-    }
+    addTwo.dependsOn(addOne)
+    addThree.dependsOn(addTwo)
+    addFour.dependsOn(addThree)
 
-    "Results are combined" - {
-      def add(x: Int): Task[Set[Int]] = new Task(_ + x, () => ())
-
-      val addOne = add(1)
-      val addTwo = add(2)
-      val addThree = add(3)
-      val addFour = add(4)
-      val addFive = add(5)
-
-      addTwo.dependsOn(addOne)
-      addThree.dependsOn(addTwo)
-      addFive.dependsOn(addThree)
-      addFive.dependsOn(addFour)
-
-      addFive.run.map {
-        case Task.Success(numbers) =>
-          assert(numbers == Set(1, 2, 3, 4, 5))
-        case _ =>
-          assert(false)
-      }
-    }
+    val _ = run(addThree)
+    assertEquals(Buffer(1, 2, 3), buf)
+  }
 
-    "Task failure reports partial results" - {
-      implicit val mergeable: Mergeable[Set[Int]] =
-        sets => sets.foldLeft(Set.empty[Int])(_ ++ _)
-
-      def add(x: Int): Task[Set[Int]] =
-        new Task(nbs => {
-          if (x == 3) throw new Exception("Boom!")
-          else nbs + x
-        }, () => ())
-
-      val addOne = add(1)
-      val addTwo = add(2)
-      val addThree = add(3)
-      val addFour = add(4)
-      val addFive = add(5)
-
-      addThree.dependsOn(addTwo)
-      addFour.dependsOn(addOne)
-      addFour.dependsOn(addThree)
-      addFive.dependsOn(addFour)
-
-      addFive.run().map {
-        case Task.Failure(partial, reasons) =>
-          assert(partial == Set(1, 2))
-          assert(reasons.map(_.getMessage) == List("Boom!"))
-        case Task.Success(_) => sys.error("The task failure did not report partial results!")
-      }
+  @Test
+  def resultsAreCombined = {
```

Comment:
Remember that you can use symbols (as in `this test checks bla`) and then have junit print it correctly with a flag :wink:

Reply:
I've been quickly playing with that, and as far as I can tell, it makes it harder to select individual method to run, so I'd prefer to keep it like it is now...